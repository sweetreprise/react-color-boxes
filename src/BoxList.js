import React, {useState} from 'react'
import Box from './Box'
import NewBoxForm from './NewBoxForm'
import { v4 as uuid } from 'uuid';

const BoxList = () => {
    const INITIAL_STATE = [
        { id: uuid(), width: 100, height: 100, bgColor: '#000000'}
    ]
    const [boxes, setBoxes] = useState(INITIAL_STATE);
    // const addBox = (width, height, bgColor) => {
    //     setBoxes(boxes => [...boxes, { id: uuid(), width, height, bgColor }])
    // }

    function addBox(width, height, bgColor) {
        setBoxes(boxes => [...boxes, { id: uuid(), width, height, bgColor }])
    }

    return (
        <div>
            <h3>BOXES!</h3>
            <NewBoxForm addBox={addBox}/>
            <div>
                {boxes.map(({id, width, height, bgColor}) => <Box 
                    key={id}
                    width={width}
                    height={height}
                    bgColor={bgColor}
                />)}
            </div>
        </div>
    )
}

export default BoxList;