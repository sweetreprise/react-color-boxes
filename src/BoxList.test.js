import React from 'react';
import { render, fireEvent, queryByTestId } from "@testing-library/react";
import BoxList from './BoxList';

it("renders component without crashing", () => {
    render(<BoxList />);
});

it("matches snapshot", () => {
    const { asFragment } = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
});