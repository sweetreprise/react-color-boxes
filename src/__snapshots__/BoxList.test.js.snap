// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      BOXES!
    </h3>
    <form>
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        max="500"
        min="100"
        name="width"
        placeholder="Width"
        type="number"
        value=""
      />
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        max="500"
        min="100"
        name="height"
        placeholder="Height"
        type="number"
        value=""
      />
      <label
        for="bgColor"
      >
        Background-color:
      </label>
      <input
        id="bgColor"
        name="bgColor"
        placeholder="background-color"
        type="color"
        value="#000000"
      />
      <button>
        Enter
      </button>
    </form>
    <div>
      <div
        class="container"
      >
        <div
          class="box"
          data-testid="box"
          style="background-color: rgb(0, 0, 0); width: 100px; height: 100px;"
        >
          <button
            data-testid="delete-button"
          >
            X
          </button>
          <span
            class="color-id"
          >
            #000000
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
